frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -80,76 +70,213 @@ export function NotebookDataPicker({
     onChangeRef.current?.(table, metadataProvider);
   };
 
+  if (isEmbedding) {
+    return (
+      <LegacyDataPicker
+        query={query}
+        stageIndex={stageIndex}
+        table={table}
+        placeholder={placeholder}
+        hasMetrics={hasMetrics}
+        isDisabled={isDisabled}
+        onChange={handleChange}
+      />
+    );
+  } else {
```

Comment:
For the embedding sdk (`isEmbeddingSdk`), we could switch to the modern data picker now as we're still in beta (and we have a task https://github.com/metabase/metabase/issues/49902 for this which we originally planned to do); I can communicate the breaking changes in the customer channel. We can retain only the iframe embedding to use the legacy picker.

cc @albertoperdomo, what do you think?

Reply:
@ranquild Nevermind, I misunderstood the name of the legacy and the modern data picker, sorry. This is already correct.